frontend/src/modules/profiles/selectors.ts

```diff
@@ -0,0 +1,26 @@
+import { createSelector } from "@reduxjs/toolkit"
+import { selectAuthCurrentUserID, selectUsersByID } from "../../selectors"
+import { RootState } from "../../store"
+import { FMSProfilePreferences } from "../../models/fms_api_models"
+
+export const selectProfileByID = (state: RootState) => state.profiles.itemsByID
+
+export const selectCurrentUserProfile = createSelector(
+    [selectAuthCurrentUserID, selectUsersByID, selectProfileByID],
+    (currentUserID, usersByID, profilesByID) => {
+        if (currentUserID && currentUserID in usersByID) {
+            return profilesByID[usersByID[Number(currentUserID)].profile]
+        }
+        return undefined
+    }
+)
+
+const DEFAULT_PREFERENCES: FMSProfilePreferences = {
+    'table.sample.page-limit': 100,
+}
+export const selectCurrentPreferences = createSelector(
+    [selectCurrentUserProfile],
+    (profile) => {
+        return profile ? profile.preferences : DEFAULT_PREFERENCES
```

Comment:
this won't protect against a missing entry in the preference object... if there is a single key missing you will still get a missing key when asked for that key... Maybe you should wrap this as selectCurrentPreference(key) and in the selector return either the profile.preferences[key] or DEFAULT_PREFERENCES[key] otherwise ...